core/components/molecules/dialog/dialog.js

```diff
@@ -34,75 +35,165 @@ const createButtonForAction = (action, index) => {
+  props.role === requiredRole ? propObject : {}
+
+class Dialog extends React.Component {
+  constructor(props) {
+    super(props)
+    this.childrenRef = React.createRef()
+  }
+
+  componentDidMount() {
+    if (this.props.role === 'form') {
+      setTimeout(() => focusOnFormInput(this.childrenRef), 0)
+    }
+  }
+
+  render() {
+    const props = this.props
+    return (
+      <Overlay contentSize={props.width} {...props}>
+        <FocusTrap persistentFocus={false} onExit={props.onClose}>
+          <DialogBox
+            width={props.width}
+            {...Automation('dialog')}
+            {...roleDependantProp(props, 'destructive', {
+              'aria-describedby': 'dialog-description'
+            })}
+            role="dialog"
+            aria-modal="true"
+            aria-labelledby="dialog-title"
+          >
+            <DialogClose>
+              <Button
+                aria-label="Close"
+                size="default"
+                appearance="link"
+                icon="close"
+                onClick={props.onClose}
+              />
+            </DialogClose>
+
+            {props.title && (
+              <DialogHeader {...Automation('dialog.title')}>
+                <DialogTitle size={props.titleElement} id="dialog-title">
+                  {props.title}
+                </DialogTitle>
+              </DialogHeader>
+            )}
+
+            <DialogBody
+              ref={this.childrenRef}
+              id="dialog-description"
+              {...Automation('dialog.body')}
+            >
+              {props.children}
+            </DialogBody>
+
+            {props.actions && (
+              <DialogFooter {...Automation('dialog.footer')}>
+                <ButtonGroup>{props.actions.map(createButtonForAction)}</ButtonGroup>
+              </DialogFooter>
+            )}
+          </DialogBox>
+        </FocusTrap>
+      </Overlay>
+    )
+  }
+}
+
+const DialogBox = styled.div`
   position: relative;
-  overflow: hidden;
-  width: ${props => props.width}px;
-  background: ${colors.base.white};
+  max-height: calc(100vh - (${spacing.xlarge} * 2));
+  display: flex;
+  flex-direction: column;
+  background-color: ${colors.base.white};
   border-radius: 3px;
   box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
 `
 
-const DialogTitleBar = styled.div`
-  display: flex;
+const DialogClose = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  
+  /* Overwirtes the color of the icons */
+  ${Button.Element} {
+    &, i, svg, path {
+      color: ${colors.base.black};
+       fill: ${colors.base.black};
+    }
+  }
+`
+
+const DialogHeader = styled.header`
   position: relative;
-  overflow: hidden;
-  padding: ${spacing.small};
-  padding-bottom: 0;
+  padding: ${spacing.small} ${spacing.large} ${spacing.xsmall} ${spacing.large};
   color: ${colors.text.default};
-  font-weight: ${fonts.weight.medium};
+  word-break: break-word;
+  white-space: pre-wrap;
+  word-wrap: break-word;
```

Comment:
I'm trusting @andresgalante to know what's going on here 😋

Reply:
This is saying: if the word is a very very very long string without spaces, then break it and wrap it into a new line. You can check the stressed example.